### ISSUE TITLE
Correct handling for adding second @mention in Tavern and Guild chats

### ISSUE THREAD

**Issue by veeeeeee** [HabitRPG/habitica#12377]
Scenario: If you write a message with a single @mention and later decide to @mention a second user, the dropdown list of names does not appear for the second @mention. 

Ideally, selecting the desired choice from dropdown menu should not lose focus of the text area. (Currently, selecting your choice requires you to re-select text area in order to continue typing).

This issue is a cleanup of #9694 with additional UI improvement suggestion. 

For anyone wanting to work on this, most of the work was done in #10638 which was never merged but can be used as a good starting point.

**Comment by Amberlisi** [comment]
I haven't looked at the code yet, but would a possible solution be to add a .focus() so that the text area is selected automatically after the user selects an entry from the dropdown list? Or is there a more elegant solution to this

**Comment by Alys** [comment]
@Amberlisi I'm not sure if that will help. From my testing, I don't think the text area loses focus.

I'm not actually sure that this issue still exists, or at least not as written here. Going back to the original issue (https://github.com/HabitRPG/habitica/issues/9694), the "second scenario" written there still seems to happen (I've copied it all below) but that's a bit different than described here. However this may be browser-dependent and it's possible that the issue as written here DOES happen, just in a different browser. I'm using Firefox at the moment. If I can make time for this, I'll try to test in different browsers and then update this issue, but I don't know if/when I'll be able to.

If you'd like to take this on, you'd be welcome to, but if you'd prefer to wait for clarification or pick a different issue, that would be understandable. :) 

**"second scenario: you want to add a second mention before the first one (one of those posts in whichc you write to multiple people and then want to address the first paragraph also to someone else), the list of names doesn't appear."""

**Comment by Alys** [comment]
Ah, actually, if you select the second item **with your mouse**, then the issue as written here still happens, so yes, focus() in that case may help! Selecting with keyboard arrow keys doesn't cause focus to be lost. The "second scenario" still exists though.

**Comment by Amberlisi** [comment]
Yeah, I can pick this up and look into it further! It doesn't seem like an urgent issue, right?

**Comment by Alys** [comment]
@Amberlisi that's right, it's not urgent. Thank you! I've marked it as in progress for you. If you run into trouble or have questions or need clarification for any of it, just post here and we'll help!

**Comment by shanaqui** [comment]
@Amberlisi Hey! Just checking in whether you're still interested in doing this fix -- not to rush you, just to make sure it doesn't go stale if you're not interested anymore. :) Let us know within a week if you want to keep working on it; otherwise, I'll mark it as help wanted again. Thanks!

**Comment by Amberlisi** [comment]
> @Amberlisi Hey! Just checking in whether you're still interested in doing this fix -- not to rush you, just to make sure it doesn't go stale if you're not interested anymore. :) Let us know within a week if you want to keep working on it; otherwise, I'll mark it as help wanted again. Thanks!

Hey @shanaqui, yeah I just got back into looking at this. Got busy with work so I had to put this one off for a bit.

**Comment by shanaqui** [comment]
@Amberlisi Thanks for checking in! :)

**Comment by shanaqui** [comment]
Hey @Amberlisi, just checking in again! Not to hurry you (it's fine if it takes some time!) but just to make sure the issue doesn't go stale. :) Let me know if you're still planning on working on this, and if not or if I don't hear from you within a week, I'll re-add the help wanted label. Thanks!

**Comment by Amberlisi** [comment]
> Hey @Amberlisi, just checking in again! Not to hurry you (it's fine if it takes some time!) but just to make sure the issue doesn't go stale. :) Let me know if you're still planning on working on this, and if not or if I don't hear from you within a week, I'll re-add the help wanted label. Thanks!

Hey, yeah I'm still working on this!

**Comment by shanaqui** [comment]
Hi @Amberlisi, just checking in again! :)

**Comment by Amberlisi** [comment]
@shanaqui Hey, if someone else is able to pick this up and work on it they should. I am dealing with being laid off just recently and need to take some time to figure that situation out.

**Comment by shanaqui** [comment]
@Amberlisi Sorry to hear that, and sorry about the late reply! Making it Help Wanted again for now, but give us a shout if you want to give it another try.

**Comment by LioZam** [comment]
I am interested helping with this issue. (I experienced it *soo* often, it bugged me XD (pun intended)).

If I am allowed to work on it, would you point me towards the relevant file/code to look at? @Amberlisi 
Thank you all for your work!

**Comment by shanaqui** [comment]
Hi @LioZam -- go for it! I'll mark this as in progress for you.

**Comment by shanaqui** [comment]
@LioZam Hello! Are you still working on this? If not, or if we don't hear back from you within a week, I'll mark it as help wanted again so the issue doesn't go totally stale. :)

**Comment by LioZam** [comment]
@shanaqui hi, no I didn't work on it since I didn't hear anything about it, back then, when I had the time for this. Sorry, I didn't catch that you changed the labels. Please go on and mark it as help wanted again. thanks.

**Comment by Helcostr** [comment]
### First scenario
I replicated the first issue by typing the names back to back without a space `@name1@name2`, but I don't think that's a bug... 
Any more perspective on browser specific issues/different formatting would be a great perspective to fix this issue.

### Second scenario
I have also replicated the second scenario using the mouse to select the user and can fix this.

**Comment by Alys** [comment]
@Helcostr Thanks! I'm marking this as in progress for you.

You're correct that when typing the names back to back without a space @name1@name2 it's not a bug that the drop-down doesn't appear.

----

I've deleted some older comments that weren't directly related to the details of the bugs and fixes to make the bug information easier to find in the remaining comments.

**Comment by Helcostr** [comment]
Ah, upon closer inspection, I discovered that the dropdown doesn't happen for the 2nd user IF I am typing in the middle of the paragraph... (so I think the first issue still exists)

Will see if I can upgrade the textarea into some kind of "div contenteditable" system...

**Comment by Alys** [comment]
@Helcostr Before you make that change, I think we should consider if it has any unintended side-effects. For example, will it still be as accessible for people using screen reader software; will it still be draggable to change the size; will keystrokes like Ctrl-Enter still work as they do now?

**Comment by Helcostr** [comment]
@Alys Well, if there isn't a library to think of those for me: yes. I never leave anything unintended if possible. This is shoved to the bottom of my priority pile, unless u want me to tackle the 2nd issue first before tackling the first (these are two separate issues that can be disentangled from each other).

Edit: Oh geez, you are right... cross browser contenteditable is hard to patch and cover, (in terms of hooking my own JS events).

**Comment by Alys** [comment]
@Helcostr Sorry about the delay! I think it would be fine to fix the second issue first if you'd like to! If you write something like `partial fix for #12377` in your PR, it will link it to this issue without automatically closing the issue.

**Comment by Helcostr** [comment]
Uh, no, I'm good. Fixing both issue 1 and two (I almost finished fixing 1), since I realized that div-contenteditable can have a ton of JS side effects (since I have to capture so many events). (Time to make an nodejs electron version of Habitica that someone can download to their desktop haha). And since I refuse to use div-contenteditable, I can fix the first issue using the textarea we got.

I am a bit weirded out that v-model isn't updating live... 🤔 so I'm just poking around seeing if I need to restructure that part. But yeah. Things are great.

Edit: Both issue 1 and 2 has been fixed. (Will attach PR)
